docs/vc-spec.md

```diff
@@ -0,0 +1,313 @@
+side can be identified using `id_alias` for the purpose of attribute sharing, and that the credential
+described by `credential_spec` does apply to the user.  When these checks are successful, the issuer
+prepares and returns a context in `PreparedCredentialData.prepared_context` (if any).  The returned prepared context is then 
+passed back to the issuer in a subsequent `get_credential`-call (see below).
+This call must be authenticated (i.e. it is not available for the anonymous sender).
+
+**NOTE:** 
+The value of `prepared_context` is basically used to transfer information between `prepare_credential`
+and `get_credential` steps, and it is totally up to the issuer to decide on the content of 
+that field.  That is, the issuer creates `prepared_context`, and is the only entity that
+consumes it.
+
+
+### 3: Get Credential
+
+`get_credential`-service issues the actual credential requested by the user.
+
+```candid
+service : {
+  get_credential : (GetCredentialRequest) -> (GetCredentialResponse) query;
+};
+
+type GetCredentialRequest = record {
+    signed_id_alias : SignedIdAlias;
+    prepared_context : opt blob;
+    credential_spec : CredentialSpec;
+};
+type GetCredentialResponse = variant {
+    Ok : IssuedCredentialData;
+    Err : IssueCredentialError;
+};
+type IssuedCredentialData = record { vc_jws : text };
+
+```
+
+`GetCredentialRequest` should contain the same parameters as `PrepareCredentialRequest`, plus the
+`prepared_context`-value  returned by `prepare_credential`, if any.
+The issuer performs the same checks as during the `prepare_credential`-call,
+plus verify that `prepared_context` is consistent with the other parameters.
+This call must be authenticated (i.e. it is not available for the anonymous sender).
+
+Upon successful checks, issuer returns the signed credential in JWS-format.
+
+
+##  Identity Provider API 
+
+This section describes the _Window Post Message_-interface implemented by the identity provider (Internet Idenity).
+This interface is used by a Relying Party during attribute sharing flow (cf. [flow description](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/attribute-sharing.md))
+
+### 1: Load II in a new window
+
+The II window needs to be opened with the subpath `/vc-flow`.
+
+After opening the II window, II will load and notify `window.opener` with the following JSON-RPC notification:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "vc-flow-ready"
+}
+```
+
+### 2: Request a VC
+
+After receiving the notification that II is ready, the relying party can request a VC by sending the following JSON-RPC request:
+ 
+* Method: `request_credential`
+* Params:
+  * `issuer`: An issuer that the relying party trusts. It has the following properties:
+    * `origin`: The origin of the issuer.
+    * `canisterId`: (optional) The canister id of the issuer, if applicable/known.
```

Comment:
what are the semantics or `cansiterId`? what if both `origin` and `canisterId` are provided?

Reply:
ok, might be worth adding something along the lines of 

> (optional) The canister id of the issuer, if applicable/known. **if specified and not the same as the one reported by the BN for `origin`, this is an error**


otherwise you could understand this as "either use the canister ID of `origin` or fallback to `canisterId` if `origin` doesn't have a canister ID"

Reply:
Added.